### PR TITLE
Add more unittests

### PR DESCRIPTION
**Pull request by fabinsch**
Additional cpp and python unittests taken from cvxpy. Also, activate python unittests in CI.

- use them to debug feasibility issue @Bambade .

### CODE REVIEW

**Review by jcarpent — APPROVED** [review]

File: include/proxsuite/proxqp/dense/utils.hpp [hunk 1]
```diff
@@ -239,7 +239,8 @@ global_primal_residual_infeasibility(VectorViewMut<T> ATdy,
   T bound_z = qpsettings.eps_primal_inf * infty_norm(dz.to_eigen());
 
   bool res = infty_norm(ATdy.to_eigen()) <= bound_y && eq_inf <= -bound_y &&
-             infty_norm(CTdz.to_eigen()) <= bound_z && in_inf <= -bound_z;
+             infty_norm(CTdz.to_eigen()) <= bound_z && in_inf <= -bound_z &&
+             infty_norm(dy.to_eigen()) !=0 && infty_norm(dz.to_eigen()) !=0;
```
Comment (jcarpent) [inline include/proxsuite/proxqp/dense/utils.hpp]: For efficiency reasons, I would first check `infty_norm(dy.to_eigen()) !=0 && infty_norm(dz.to_eigen()) !=0;` and return the solution if this statement if false.
@Bambade I have a question concerning the unscaling of the variables `ATdy`, `CTdy`, etc. Are they used outside of `global_primal_residual_infeasibility`?

**Review by jcarpent — CHANGES_REQUESTED** [review]

File: include/proxsuite/proxqp/dense/utils.hpp (hunk 1 above)
Comment (Bambade) [reply include/proxsuite/proxqp/dense/utils.hpp]: OK, it is better indeed.  
The unscaled ATdy, CTdy are not used outside of the function (the scaled versions are used  inside).

File: include/proxsuite/proxqp/dense/utils.hpp [hunk 2]
```diff
@@ -239,7 +239,8 @@ global_primal_residual_infeasibility(VectorViewMut<T> ATdy,
   T bound_z = qpsettings.eps_primal_inf * infty_norm(dz.to_eigen());
 
   bool res = infty_norm(ATdy.to_eigen()) <= bound_y && eq_inf <= -bound_y &&
```
Comment (fabinsch) [inline include/proxsuite/proxqp/dense/utils.hpp:241]:
```suggestion
  bool res = infty_norm(dy.to_eigen()) != 0 && infty_norm(dz.to_eigen()) != 0;

```

File: include/proxsuite/proxqp/dense/utils.hpp [hunk 3]
```diff
@@ -239,7 +239,8 @@ global_primal_residual_infeasibility(VectorViewMut<T> ATdy,
...
-             infty_norm(CTdz.to_eigen()) <= bound_z && in_inf <= -bound_z;
+             infty_norm(CTdz.to_eigen()) <= bound_z && in_inf <= -bound_z &&
+             infty_norm(dy.to_eigen()) != 0 && infty_norm(dz.to_eigen()) != 0;
```
Comment (fabinsch) [inline include/proxsuite/proxqp/dense/utils.hpp:243]:
```suggestion
  if (!res) {
    return res;
  } else {
    res = infty_norm(ATdy.to_eigen()) <= bound_y && eq_inf <= -bound_y &&
          infty_norm(CTdz.to_eigen()) <= bound_z && in_inf <= -bound_z;
  }
```

File: include/proxsuite/proxqp/dense/utils.hpp (hunk 1 above)
Comment (fabinsch) [reply include/proxsuite/proxqp/dense/utils.hpp]: I did a suggestion of how we could change it accordingly, first checking `nfty_norm(dy.to_eigen()) !=0 && infty_norm(dz.to_eigen()) !=0;`

File: include/proxsuite/proxqp/dense/utils.hpp [hunk 4]
```diff
@@ -227,20 +227,26 @@ global_primal_residual_infeasibility(VectorViewMut<T> ATdy,
...
+               positive_part(-dz.to_eigen()).dot(qpwork.l_scaled);
+    ruiz.unscale_dual_in_place_eq(dy);
+    ruiz.unscale_dual_in_place_in(dz);
+
+    T bound_y = qpsettings.eps_primal_inf * infty_norm(dy.to_eigen());
+    T bound_z = qpsettings.eps_primal_inf * infty_norm(dz.to_eigen());
+
+    res = infty_norm(ATdy.to_eigen()) <= bound_y && eq_inf <= -bound_y &&
+          infty_norm(CTdz.to_eigen()) <= bound_z && in_inf <= -bound_z;
+  }
```
Comment (jcarpent) [inline include/proxsuite/proxqp/dense/utils.hpp:249]: You should remove the else here and put the content in the main scope of the function.

**Pull request merged**